### ISSUE TITLE
training single speaker different accent/language results

### ISSUE THREAD

**Issue by tosunozgun** [yl4579/StyleTTS2#185]
Hi,
 I trained model with turkish speech dataset like 20 hours(clips length 1.5 - 11 sec). Started from training pl-bert after that i make 2 experiments:
1- Finetune from the exist model and use my own pl-bert
2- Train from start with my pl-bert

Used default configs in library, working on 8xA100 80GB system (for second stage training used 2 gpus cause its throws error).

The weird thing is finetuned model output performs better but not good. I will share results and train.log.

For pl-bert training i dont train it for long time(cause loss not decreased) maybe i make wrong things when changing code to adapt it for turkish, but this not explains me why finetuned model working better. 

Here is results:

[train.log](https://github.com/yl4579/StyleTTS2/files/13873694/train.log)

finetune:

https://jumpshare.com/v/m0IEZ6iUXMAp1k1VTsv7?b=SKK2pZUnMgOovElkvwgB

train:

https://jumpshare.com/v/iI23FoXaN3jXqyeJ3W5E?b=SKK2pZUnMgOovElkvwgB

**Comment by martinambrus** [comment]
How did you train 2nd stage on 2 GPUs? I could only do it on a single GPU, since there is no DDP support for it.
As for the output of 2nd stage training - it sounds to me like it was trained for maybe a couple of epochs. How many epochs did you train your 1st and 2nd phases for?

I'll be trying to train my own PL-BERT for Slovak language later myself, so I'd appreciate feedback from others in this issue as well :)

**Comment by tosunozgun** [comment]
Updated result audios.

> How did you train 2nd stage on 2 GPUs? I could only do it on a single GPU, since there is no DDP support for it. As for the output of 2nd stage training - it sounds to me like it was trained for maybe a couple of epochs. How many epochs did you train your 1st and 2nd phases for?

I know that repo has problems on training second stage with multi gpu, but when i try to train it on single gpu(my env is 8x a100(80gb)) get OOM error at 20. epoch it was starting point on diff_epoch. just giving export CUDA_VISIBLE_DEVICES=0,1 solves OOM error and model consuming around 70gb memory each of card and the second stage logs is on my post.

I used the default config.yml and some settings is below.
save_freq: 2
log_interval: 10
device: "cuda"
epochs_1st: 200 # number of epochs for first stage training (pre-training)
epochs_2nd: 100 # number of peochs for second stage training (joint training)
batch_size: 16
max_len: 400# maximum number of frames
.....
loss_params:
    lambda_mel: 5. # mel reconstruction loss
    lambda_gen: 1. # generator loss
    lambda_slm: 1. # slm feature matching loss
    
    lambda_mono: 1. # monotonic alignment loss (1st stage, TMA)
    lambda_s2s: 1. # sequence-to-sequence loss (1st stage, TMA)
    TMA_epoch: 50 # TMA starting epoch (1st stage)

    lambda_F0: 1. # F0 reconstruction loss (2nd stage)
    lambda_norm: 1. # norm reconstruction loss (2nd stage)
    lambda_dur: 1. # duration loss (2nd stage)
    lambda_ce: 20. # duration predictor probability output CE loss (2nd stage)
    lambda_sty: 1. # style reconstruction loss (2nd stage)
    lambda_diff: 1. # score matching loss (2nd stage)
    
    diff_epoch: 20 # style diffusion starting epoch (2nd stage)
    joint_epoch: 50 # joint training starting epoch (2nd stage)

optimizer_params:
  lr: 0.0001 # general learning rate
  bert_lr: 0.00001 # learning rate for PLBERT
  ft_lr: 0.00001 # learning rate for acoustic modules

**Comment by tosunozgun** [comment]
I have successfully completed the model training and the outputs are truly satisfactory. Thank you to everyone who contributed to the project.

**Comment by tuanh123789** [comment]
hi @tosunozgun @martinambrus . Face the same problem when training 2st stage in vietnamese. Output is noisy, do training more epochs can solve this
